app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -1218,25 +1233,43 @@ class SectionRenderingService(
         new components.GovukRadios(govukErrorMessage, govukFieldset, govukHint, govukLabel)(radios)
 
       case Checkbox =>
-        val items = optionsWithHintAndHelpText.zipWithIndex.map { case ((option, maybeHint, maybeHelpText), index) =>
-          CheckboxItem(
-            id = Some(formComponent.id.value + index),
-            value = index.toString,
-            content = content.Text(option.value),
-            checked = isChecked(index),
-            conditionalHtml = helpTextHtml(maybeHelpText),
-            attributes = dataLabelAttribute(option),
-            hint = maybeHint
-          )
-        }
+        val items = dividerPosition.foldLeft(optionsWithHintAndHelpText.zipWithIndex.map {
+          case ((option, maybeHint, maybeHelpText), index) =>
+            if (noneChoice == Some(index + 1)) {
+              CheckboxItem(
+                id = Some(formComponent.id.value + index),
+                value = index.toString,
+                content = content.Text(option.value),
+                checked = isChecked(index),
+                conditionalHtml = helpTextHtml(maybeHelpText),
+                attributes = dataLabelAttribute(option),
+                hint = maybeHint,
+                behaviour = Some(ExclusiveCheckbox)
+              )
+            } else {
+              CheckboxItem(
+                id = Some(formComponent.id.value + index),
+                value = index.toString,
+                content = content.Text(option.value),
+                checked = isChecked(index),
+                conditionalHtml = helpTextHtml(maybeHelpText),
+                attributes = dataLabelAttribute(option),
+                hint = maybeHint
+              )
+            }
```

Comment:
There is quite a duplication here. What about to use 
```suggestion
          val item = CheckboxItem(
            id = Some(formComponent.id.value + index),
            value = index.toString,
            content = content.Text(option.value),
            checked = isChecked(index),
            conditionalHtml = helpTextHtml(maybeHelpText),
            attributes = dataLabelAttribute(option),
            hint = maybeHint
          )
          if (noneChoice.contains(index + 1)) {
            item.copy(behaviour = Some(ExclusiveCheckbox))
          } else {
            item
          }
```